superset-frontend/src/views/components/MenuRight.tsx

```diff
@@ -113,9 +148,41 @@ const RightMenu = ({
             }
             icon={<Icons.TriangleDown />}
           >
-            {dropdownItems.map(
-              menu =>
-                findPermission(menu.perm, menu.view, roles) && (
+            {dropdownItems.map(menu => {
+              if (menu.label === 'Data') {
+                return (
+                  <SubMenu
+                    key="sub2"
+                    className="data-menu"
+                    title={menuIconAndLabel(menu as MenuObjectProps)}
+                  >
+                    {allowedExtensions?.csv_extensions && (
```

Comment:
similarly here, it looks like we have the text for each of these menus in the childs array.. can we loop over that array here and pull the data from above instead of hard coding it in the render function? That may be dry-er. We could also add the permissions to look up in the arrays above.

Reply:
Would it make sense to hard code these because I have to check the allowedExtensions bools? Otherwise I'll have to add a method and hash to check if to render these.